local/scripts/install_tools.sh

```diff
@@ -1,113 +1,259 @@
 #!/usr/bin/env bash
 
-# This script installs Docker, Kind, Helm, and Tilt if they are not already installed.
-# It logs each step and provides a basic explanation of how functions work via comments.
+# This script installs Docker, Kind, Kubectl, Helm, and Tilt if they are not already installed.
+# It detects the OS and architecture to download the correct binaries.
+
+set -e
+
+# Terminal colors
+RED='\033[0;31m'
+GREEN='\033[0;32m'
+YELLOW='\033[0;33m'
+BLUE='\033[0;34m'
+RESET='\033[0m'
+
+# Function to log messages to file and console
+log() {
+    local level="$1"
+    local message="$2"
+    local color="$RESET"
+    
+    case "$level" in
+        "INFO") color="$BLUE" ;;
+        "SUCCESS") color="$GREEN" ;;
+        "WARNING") color="$YELLOW" ;;
+        "ERROR") color="$RED" ;;
+    esac
+    
+    echo -e "${color}${message}${RESET}"
+}
 
 # Function to check if a command exists on the system
-# This function takes a single argument (the command name), checks if it's available, and returns 0 if found, 1 if not.
 command_exists() {
     command -v "$1" >/dev/null 2>&1
 }
 
+# Function to detect system architecture and OS
+detect_system() {
```

Comment:
Same as above comment: I suggest looking for off-the-shelf alternatives here.

Reply:
Ditto. TODO added to use single image with all deps and remove the need for the script altogether.